src/main/java/com/redhat/pantheon/servlet/ModuleRenderServlet.java

```diff
@@ -0,0 +1,121 @@
+package com.redhat.pantheon.servlet;
+
+import com.redhat.pantheon.model.module.Metadata;
+import com.redhat.pantheon.model.module.Module;
+import com.redhat.pantheon.model.module.Content;
+import com.redhat.pantheon.model.module.ModuleRevision;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.servlets.annotations.SlingServletPaths;
+
+import org.jetbrains.annotations.NotNull;
+
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Component;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.Servlet;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.redhat.pantheon.conf.GlobalConfig.DEFAULT_MODULE_LOCALE;
+import static com.redhat.pantheon.conf.GlobalConfig.CONTENT_TYPE;
+import static com.redhat.pantheon.servlet.ServletUtils.paramValue;
+
+/**
+ * Get operation to render a Module data in JSON format.
+ * Only two parameters are expected in the Get request:
+ * 1. locale - Optional; indicates the locale that the module content is in, defaulted to en-US
+ * 2. module_id - indicates the uuid string which uniquely identifies a module
+ *
+ * The url to GET a request from the server is /api/module/render
+ * Example: <server_url>/api/module/render?locale=en-us&module_id=xyz
+ * The said url is accessible outside of the system without any authentication.
+ *
+ * @author Ankit Gadgil
+ */
+@Component(
+        service = Servlet.class,
+        property = {
+                Constants.SERVICE_DESCRIPTION + "=Servlet to facilitate GET operation which accepts locale and module uuid to output module data",
+                Constants.SERVICE_VENDOR + "=Red Hat Content Tooling team"
+        })
+@SlingServletPaths(value = "/api/module/render")
+public class ModuleRenderServlet extends AbstractJsonSingleQueryServlet {
```

Comment:
This is a naming standard based on the initial decision. I agree with Ben that it might be an inaccurate name. Now is the time to change the naming if we are going to do so.

Reply:
Renaming the servlet makes sense and I am all for it.